modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java

```diff
@@ -142,27 +246,79 @@ public void testUpdateObjectValidationRule() throws Exception {
+				objectValidationRule1.getObjectValidationRuleId(), true,
 				ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
 				LocalizedMapUtil.getLocalizedMap("Field must be an URL"),
 				LocalizedMapUtil.getLocalizedMap("URL Validation"),
 				ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION,
 				"isURL(textField)", Collections.emptyList());
 
-		Assert.assertTrue(objectValidationRule.isActive());
+		Assert.assertTrue(objectValidationRule1.isActive());
 		Assert.assertEquals(
 			ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
-			objectValidationRule.getEngine());
+			objectValidationRule1.getEngine());
 		Assert.assertEquals(
 			LocalizedMapUtil.getLocalizedMap("Field must be an URL"),
-			objectValidationRule.getErrorLabelMap());
+			objectValidationRule1.getErrorLabelMap());
 		Assert.assertEquals(
 			LocalizedMapUtil.getLocalizedMap("URL Validation"),
-			objectValidationRule.getNameMap());
+			objectValidationRule1.getNameMap());
 		Assert.assertEquals(
-			"isURL(textField)", objectValidationRule.getScript());
+			"isURL(textField)", objectValidationRule1.getScript());
+
+		List<ObjectField> objectFields =
+			_objectFieldLocalService.getObjectFields(
+				_objectDefinition.getObjectDefinitionId(), false);
+
+		ObjectField objectField1 = objectFields.get(0);
+		ObjectField objectField2 = objectFields.get(1);
+
+		ObjectValidationRule objectValidationRule2 =
+			_objectValidationRuleLocalService.addObjectValidationRule(
+				TestPropsValues.getUserId(),
+				_objectDefinition.getObjectDefinitionId(), true,
+				ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectValidationRuleConstants.OUTPUT_TYPE_PARTIAL_VALIDATION,
+				_VALID_DDM_SCRIPT,
+				Collections.singletonList(
+					_getObjectValidationRuleSetting(
+						ObjectValidationRuleSettingConstants.
+							NAME_OBJECT_FIELD_ID,
+						String.valueOf(objectField1.getObjectFieldId()))));
+
+		Assert.assertTrue(objectValidationRule2.isActive());
+		_assertObjectValidationRuleSettingsValues(
+			HashMapBuilder.put(
+				ObjectValidationRuleSettingConstants.NAME_OBJECT_FIELD_ID,
+				String.valueOf(objectField1.getObjectFieldId())
+			).build(),
+			objectValidationRule2.getObjectValidationRuleSettings());
+
+		objectValidationRule2 =
+			_objectValidationRuleLocalService.updateObjectValidationRule(
+				objectValidationRule2.getObjectValidationRuleId(), false,
+				objectValidationRule2.getEngine(),
+				objectValidationRule2.getErrorLabelMap(),
+				objectValidationRule2.getNameMap(),
+				objectValidationRule2.getOutputType(),
+				objectValidationRule2.getScript(),
+				Collections.singletonList(
+					_getObjectValidationRuleSetting(
+						ObjectValidationRuleSettingConstants.
+							NAME_OBJECT_FIELD_ID,
+						String.valueOf(objectField2.getObjectFieldId()))));
+
+		Assert.assertFalse(objectValidationRule2.isActive());
```

Comment:
I believe that we don't need this assert here.

Reply:
This might be necessary to assure that the validation is actually being deactivated. A similar assert is done [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java#L154)